src/tsickle.ts

```diff
@@ -2013,189 +2021,75 @@ export function emitWithTsickle(
-        // classes need special care to match clutz, which seperates class types into a
-        // type for the static properties and a type for the instance properties
-        reexports.push({
-          namespace: 'ಠ_ಠ.clutz',
-          clutzSymbolName: `module$contents$${clutzModuleName}_${symbol.name}_Instance`,
-          aliasedSymbolName: symbol.name
-        });
-        reexports.push({
-          namespace: 'ಠ_ಠ.clutz.module$exports$' + clutzModuleName,
-          clutzSymbolName: symbol.name + '_Instance',
-          aliasedSymbolName: `module$contents$${clutzModuleName}_${symbol.name}`
-        });
-      }
-
-      if (symbol.flags & ts.SymbolFlags.Type || symbol.flags & ts.SymbolFlags.Class) {
-        for (const {namespace, clutzSymbolName, aliasedSymbolName} of reexports) {
-          addToMultiMap(
-              reexportsByNamespace, namespace,
-              `type ${clutzSymbolName}${paramsWithContraint} = ${aliasedSymbolName}${params};`);
-        }
-      }
-      if (symbol.flags & ts.SymbolFlags.Value || symbol.flags & ts.SymbolFlags.Class) {
-        for (const {namespace, clutzSymbolName, aliasedSymbolName} of reexports) {
-          addToMultiMap(
-              reexportsByNamespace, namespace,
-              `const ${clutzSymbolName}: typeof ${aliasedSymbolName};`);
-        }
-      }
+  const moduleSymbol = typeChecker.getSymbolAtLocation(sourceFile);
+  const moduleExports = moduleSymbol && typeChecker.getExportsOfModule(moduleSymbol);
+  if (!moduleExports) return dtsFileContent;
+
+  // .d.ts files can be transformed, too, so we need to compare the original node below.
+  const origSourceFile = ts.getOriginalNode(sourceFile);
+  // The module exports might be re-exports, and in the case of "export *" might not even be
+  // available in the module scope, which makes them difficult to export. Avoid the problem by
+  // filtering out symbols who do not have a declaration in the local module.
+  const localExports = moduleExports.filter(e => {
+    // If there are no declarations, be conservative and emit the aliases.
+    if (!e.declarations) return true;
+    // Skip default exports, they are not currently supported.
+    // default is a keyword in typescript, so the name of the export being default means that it's a
+    // default export.
+    if (e.name === 'default') return false;
+    // Otherwise check that some declaration is from the local module.
+    return e.declarations.some(d => d.getSourceFile() === origSourceFile);
+  });
+  if (!localExports.length) return dtsFileContent;
+
+  const moduleName = host.pathToModuleName('', sourceFile.fileName);
+  const clutzModuleName = moduleName.replace(/\./g, '$');
+
+  // Clutz might refer to the name in two different forms (stemming from goog.provide and
+  // goog.module respectively).
+  // 1) global in clutz:   ಠ_ಠ.clutz.module$contents$path$to$module_Symbol...
+  // 2) local in a module: ಠ_ಠ.clutz.module$exports$path$to$module.Symbol ..
```

Comment:
Is the module_Symbol vs module.Symbol distinction here important?  I don't see the underscores in the Clutz test suite.

Reply:
I find stuff like:

`src/test/java/com/google/javascript/clutz/partial/override.d.ts:  class module$contents$override_Base extends module$contents$override_Base_Instance {`

I think this is an artifact of Closure's `goog.provide` vs `goog.module` stuff, but tbh I don't quite know which is which. I'm aping the previous emit, and it seems to work in google3.